### ISSUE TITLE
EscapePolicy::Basics escapes unicode

### ISSUE THREAD

**Issue by johansigfrids** [zonyitoo/rust-ini#64]
It seems both `EscapePolicy::Basics` and `EscapePolicy::BasicsUnicode` escape non-ascii unicode characters. `EscapePolicy::Nothing` does not escape unicode, but doesn't escape anything else either. 

For example, have a `in.ini` files with some unicode in it:
```ini
some-key=åäö
```

And a `main.rs` file which reads and writes the ini:
```rust
use ini::Ini;
use ini::ini::EscapePolicy;

fn main() {
    let ini = Ini::load_from_file("in.ini").unwrap();
    ini.write_to_file_policy("basics.ini", EscapePolicy::Basics).unwrap();
    ini.write_to_file_policy("basics_unicode.ini", EscapePolicy::BasicsUnicode).unwrap();
}
```

Both `basics.ini` and `basics_unicode.ini` end up the same:
```ini
some-key=\x00e5\x00e4\x00f6

```

I would have expected `basics.ini` to leave the unicode stuff as is.

**Comment by zonyitoo** [comment]
Because `åäö` are `\u{e5}`, \u{e4}`, `\u{f6}`, which are in the range `\x7f`..=`\u{00ff}`, so they are controlled by `escape_basics()`.

So if you don't want them to be printed as original, use `EscapePolicy::Nothing`.

**Comment by johansigfrids** [comment]
If `EscapePolicy::Basics` escapes unicode, then what does `EscapePolicy::BasicsUnicode` do differently?

**Comment by zonyitoo** [comment]
I checked all commits related to this line of code, and it seems that it is bug introduced 4 years ago.

4 years ago @mitsuhiko added `EscapePolicy` and simply replaced the original escape match condition with a check `escape_basics()`.

**Comment by zonyitoo** [comment]
@johansigfrids Please verify if it this commit works in your case.

**Comment by johansigfrids** [comment]
Yes! Now `EscapePolicy::Basics ` does not escape the unicode, but `EscapePolicy::BasicsUnicode` does.

**Comment by zonyitoo** [comment]
> Yes! Now `EscapePolicy::Basics ` does not escape the unicode, but `EscapePolicy::BasicsUnicode` does.

It is expected, right?

If everything is ok, I am going to release a new version.

**Comment by johansigfrids** [comment]
Yes, it now works the way I would expect it to.

**Comment by zonyitoo** [comment]
Released in v0.15.2.
